components/azure/azuresynapse/runsparkjob/README.md

```diff
@@ -0,0 +1,71 @@
+- You can choose to return after the job is scheduled, or wait until the just finished.
+
+## Input Parameters
+### Job scheduling parameters
+| Name | Type | Required | Description |
+| -------------------------- | ------ | -------- | ----------------------------------------|
+| executor_size | String | Y | size of executors. Accepted values: Large, Medium, Small |
+| executors | Integer | Y | number of executors |
+| main_class_name | String | Y | The fully-qualified identifier or the main class that is in the main definition file |
+| main_definition_file | String | Y | The main file used for the job |
+| name | String | Y | The Spark job name |
+| spark_pool_name | String | Y | The Spark pool name |
+| workspace_name | String | Y | The Synapse workspace name |
+| subscription_id | String | Y | The id of Azure subscription where the Synapse workspace is in |
+| resource_group | String | Y | The Azure resource group where the Synapse workspace is in |
+| command_line_arguments | String | N | The command line arguments for the job |
+| configuration | String | N | The configuration of Spark job |
+| language | String | N | The language of Spark job. Accepted values: CSharp, PySpark, Python, Scala, Spark, SparkDotNet |
+| reference_files | String | N | Additional files used for reference in the main definition file |
+| tags | String | N | Space-separated tags: key[=value] [key[=value] ...] |
+| spark_pool_config_file | String | N | Path of the spark pool configuration yaml file. Default value is ./src/spark_pool_config.yaml |
+| wait_until_job_finished | Bool | N | Whether wait for the job completion. Default value is True |
+| waiting_timeout_in_seconds | Integer | N | The waiting timeout in seconds. Default value is 3600 |
+
+### Authentication parameters
+| Name | Type | Required | Description |
+| -------------------------- | ------ | -------- | ----------------------------------------|
+| service_principal_id | String | Y | The service principal client id |
+| service_principal_password | String | Y | The service principal password/secret |
+| tenant_id | String | Y | The Azure tenant id for the service principal |
+
+## Prerequisites
+- Create an AKS cluster
+- Install Kubeflow on AKS following [instructions for deploying Kubeflow on Azure](https://www.kubeflow.org/docs/azure/).
+- Create AAD service principal (TODO: document details)
+- Create a Azure Synapse Workspace and grant following permissions to the service principal:
+  - Azure Owner or Contributor for the Synapse workspace
+  - bigDataPools/useCompute in Synapse RBAC
+  - Storage blob data owner for the attached ADSL account (storing job 
+  definition file)
+
+  See [Synapse RBAC Roles documentation](https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-understand-what-role-you-need) for more details
+- Create a Azure Container Registry and grant access to the AKS cluster (TODO: document steps)
```

Comment:
Please link to these instructions:
https://docs.microsoft.com/azure/aks/cluster-container-registry-integration

Reply:
Though if we can eliminate the need for building the container ourselves that would be even better (and then we don't need to set up ACR and auth to ACR)

Reply:
will discuss the manage container image in other comments below.